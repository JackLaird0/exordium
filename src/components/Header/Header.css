header {
  height: 90px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 100px;
  border-bottom: 2px solid #2D0480
}

.logo {
  height: 70px;
}

.nav {
  text-decoration: none;
}

.login,
.signup {
  height: 50px;
  width: 100px;
  border-radius: 4px;
  font-size: 20px;
  cursor: pointer;
  box-shadow: 2px 3px 3px #9E9695;
  transition: 0.3s;
  font-family: 'Open Sans', sans-serif;
}

.signup {
  background: linear-gradient(to right, #C86DD7, #1A0480 );
  background-repeat: no-repeat;
  color: white;
  margin-left: 50px;
}

.login {
  border: solid 2px #2D0480;
  color: #2D0480;
}

.login:hover,
.signup:hover {
  box-shadow: 1px 2px 2px #9E9695;
}