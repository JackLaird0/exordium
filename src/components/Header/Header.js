import React, { Component } from 'react';
import './Header.css';
import { NavLink } from 'react-router-dom';
import logo from '../../assets/exordium-logo.png';

export class Header extends Component {
  render() {
    return (
      <header>
        <div className="title-logo">
          <NavLink to='/'>
            <img className="logo" src={logo} alt="exordium logo"/>
          </NavLink>
        </div>
        <div className="signin-container">
          <button className="signin-button" onClick={this.props.showLogin}>Sign Up/Login</button>
        </div>
      </header>
    )
  }
}

export default Header;