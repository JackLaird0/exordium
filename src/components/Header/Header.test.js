import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header', () => {

  it('should match the snapshot', () => {
    let wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot();
  })
})