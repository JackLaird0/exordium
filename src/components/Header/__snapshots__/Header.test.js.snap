// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title-logo"
>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/"
          >
                    <img
                              alt="exordium logo"
                              className="logo"
                              src="exordium-logo.png"
                    />
          </NavLink>
</div>,
        <div
          className="signin-container"
>
          <button
                    className="signin-button"
                    onClick={undefined}
          >
                    Sign Up / Login
          </button>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            aria-current="page"
            to="/"
>
            <img
                        alt="exordium logo"
                        className="logo"
                        src="exordium-logo.png"
            />
</NavLink>,
          "className": "title-logo",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <img
              alt="exordium logo"
              className="logo"
              src="exordium-logo.png"
/>,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "exordium logo",
              "className": "logo",
              "src": "exordium-logo.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="signin-button"
            onClick={undefined}
>
            Sign Up / Login
</button>,
          "className": "signin-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up / Login",
            "className": "signin-button",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Sign Up / Login",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title-logo"
>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/"
            >
                        <img
                                    alt="exordium logo"
                                    className="logo"
                                    src="exordium-logo.png"
                        />
            </NavLink>
</div>,
          <div
            className="signin-container"
>
            <button
                        className="signin-button"
                        onClick={undefined}
            >
                        Sign Up / Login
            </button>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              to="/"
>
              <img
                            alt="exordium logo"
                            className="logo"
                            src="exordium-logo.png"
              />
</NavLink>,
            "className": "title-logo",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": <img
                alt="exordium logo"
                className="logo"
                src="exordium-logo.png"
/>,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "exordium logo",
                "className": "logo",
                "src": "exordium-logo.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="signin-button"
              onClick={undefined}
>
              Sign Up / Login
</button>,
            "className": "signin-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up / Login",
              "className": "signin-button",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "Sign Up / Login",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
